src/dynamodb.ts

```diff
@@ -0,0 +1,93 @@
+import { DynamoDB } from "aws-sdk";
+import { ClientConfiguration, ScanInput } from "aws-sdk/clients/dynamodb";
+import { DateTime } from "luxon";
+import { dynamoLogger } from "./handler";
+// @ts-ignore
+// tslint:disable-next-line:no-var-requires
+const AWSXRay = require("aws-xray-sdk");
+
+export class Dynamo {
+    private readonly config: ClientConfiguration;
+    private readonly tableName: string;
+    private readonly branch: string;
+    private readonly now: DateTime = DateTime.utc();
+    private readonly numOfScanners: number;
+
+    public constructor(config?: ClientConfiguration, branch?: string, numOfScanners: number = 4) {
+        this.config = config || { region: process.env.AWS_REGION || "eu-west-1" };
+        this.branch = branch || process.env.BRANCH || "local";
+        this.tableName = `cvs-${this.branch.toLowerCase()}-activities`;
+        this.numOfScanners = numOfScanners;
+    }
+
+    /**
+     * Returns the date formatted to CVS project standard.
+     * @param {DateTime} date A Luxon Datetime.
+     * @returns {string}
+     */
+    public static async toCVSDate(date: DateTime): Promise<string> {
+        return await date.toISO({ includeOffset: false }) + "Z";
+    }
+
+    /**********************************************************
+     * Runs a scan against a DynamoDB table specified in {@link ScanInput}.
+     * @private
+     * @see getVisits()
+     * @see getOldVisits()
+     * @param {ScanInput} query The scan input for the request.
+     * @returns {Promise<number>} Count of records from the scan.
+     */
+    private async scan(query: ScanInput): Promise<number> {
+        const client: DynamoDB = AWSXRay.captureAWSClient(new DynamoDB(this.config));
+        const scanners: Array<Promise<number>> = [];
+        while (scanners.length < this.numOfScanners) {
+            const scanInput = query;
+            scanInput.Segment = scanners.length;
+            scanInput.TotalSegments = this.numOfScanners;
+            scanners.push(client.scan(scanInput).promise().then((r) => r.Count || 0));
```

Comment:
Is the assumption here that the tables will always be small? Otherwise a query might make more sense than parallel scans

Reply:
Query would require specifying PK value when the requirement is to find any and all visits regardless of PK.